modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java

```diff
@@ -522,10 +528,19 @@ private CommerceOrder _updateNestedResources(
 		OrderItem[] orderItems = order.getOrderItems();
 
 		if (orderItems != null) {
+			Long[] longs = _getOrderItemIds(orderItems);
+			String[] strings = _getOrderItemExternalReferenceCodes(orderItems);
+
+			if (ArrayUtil.isEmpty(longs)) {
+				longs = new Long[] {0L};
+			}
+
+			if (ArrayUtil.isEmpty(strings)) {
+				strings = null;
+			}
+
```

Comment:
- I think this is not needed the empty cases are already managed with the change made in the local service
- I'm not sure assign null to strings won't cause any trouble to the service
- I'm not sure put 0 as default value for id is right